WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift

```diff
@@ -45,6 +45,8 @@ class SiteIntentViewController: CollapsableHeaderViewController {
 
         tableView.dataSource = self
         searchBar.delegate = self
+
+        SiteIntentData.clearCustomVerticals()
```

Comment:
how about we put this inside `deinit()`? This way, that rogue custom vertical would not be lingering before we call it again. Or, as we discussed on Slack, we can inject this type in the site intent flow and avoid global properties, but this can be done in another PR.

Reply:
Thanks @Gio2018! This inspired me to do some refactoring with my latest set of commits.

Hopefully the benefits are:
- We no longer need to explicitly clear custom verticals
- State is contained to the Site Intent VC